app/controllers/products_controller.rb

```diff
@@ -0,0 +1,177 @@
+      @product.update(:price => params[:product][:price])
+      @product.update(:inventory => params[:product][:inventory])
+      @product.update(:description => params[:product][:description])
+      @product.photo_url.empty?? @product.update(:photo_url =>  "http://www.spotahome.com/blog/wp-content/plugins/wd-instagram-feed/images/missing.png") : @product.update(:photo_url => params[:product][:photo_url])
+      @product.retired == true ? @product.update(retired: false) : @product.update(retired: true)
+
+      @categories.each do |cat|
+        @search = @existingcats.where(category_id: cat.id)
+        if params[:product][:"#{cat.name}"] == "1" && @search.empty?
+          @prodcat = ProductCategory.new
+          @prodcat.product_id = @product.id
+          @prodcat.category_id = cat.id
+          @prodcat.save
+        elsif params[:product][:"#{cat.name}"] == "0" && @search.any?
+          ProductCategory.destroy(@search)
+        end
+      end
+      redirect_to user_product_path(params[:user_id],params[:id])
+    else
+      render :new
+    end
+  end
+
+  def category
+    @browser = BROWSE
+    @category = Category.find(params[:id])
+    @show = @category.products
+    render :categoryproducts
+  end
+
+  def destroy
+    @user = User.find(params[:user_id])
+    @product = Product.find(params[:id])
+    @product.destroy
+    redirect_to user_products_path(@user)
+  end
+
+  def retire
+    @user = User.find(params[:user_id])
+    @product = Product.find(params[:id])
+    @product.retired == false ? @product.update(retired: true) : @product.update(retired: false)
+    @product.save
+    redirect_to user_products_path(params[:user_id])
+    # redirect_to user_products_path(@user)
+  end
+
+  def show
+    @product = Product.find(params[:id])
+    @review = Review.where(product_id: @product.id)
+    @cart_item = CartItem.new
+    # @cart_items = CartItem.all
+    @item_exists_in_cart = CartItem.where(session_id: session[:session_id], product_id: params[:id])
+  end
+
+  def new_cats
+    @new_cat = Category.new
+    @categories = Category.all.order(name: :asc)
+
+  end
+
+  def update_new_cats
+    @new_cat = Category.new
+    # @categories = Category.all.order(name: :asc)
+    @new_cat.update(:name => params[:new])
+    if @new_cat.save
+      redirect_to root_path
+    end
+  end
+
+
+  # def new_cats
+  #   @new_cat = Category.new
+  #   @categories = Category.all.order(name: :asc)
+  # end
+
+  def create_cart
+    @cart_item = CartItem.new(cart_params)
+    @product = Product.find(params[:id]) || Product.find(@product.id)
+    if @cart_item.save
+      @cart_item.update(session_id: session[:session_id])
+      @cart_item.update(user_id: session[:user_id])
+      redirect_to product_path(@product.id), :notice => 'Products added to cart.'
+    else
+      render :show
+    end
+  end
+
+  private
+
+
+  def cart_params
+    params.permit(:quantity, :product_id)
+  end
+
+  private
+  def product_create_params
```

Comment:
From @lacuchilla:
I'm seeing a product_create_params private method and a product_update params private method that contain the same information. There should be a way to condense these two methods into one method and name it product_params, and then pass the product_params in the create and update methods.